src/table/utils/handle-key-press.js

```diff
@@ -148,27 +149,37 @@ export const bodyHandleKeyPress = ({
     // Space bar: Selects value.
     case ' ':
       preventDefaultBehavior(evt);
-      cell.isSelectable && isAnalysisMode && selectionDispatch({ type: 'select', payload: { cell, evt, announce } });
+      cell.isSelectable &&
+        isSelectionsEnabled &&
+        selectionDispatch({ type: 'select', payload: { cell, evt, announce } });
       break;
     // Enter: Confirms selections.
     case 'Enter':
       preventDefaultBehavior(evt);
-      if (!isSelectionMode) break;
-      selectionsAPI.confirm();
-      announce({ keys: ['SNTable.SelectionLabel.SelectionsConfirmed'] });
+      if (isSelectionMode) {
```

Comment:
I found the early returns here hard to read, I can change back if you don't agree

Reply:
your change here is good